crates/supervisor/core/src/supervisor.rs

```diff
@@ -210,8 +212,17 @@ impl SupervisorService for Supervisor {
         self.config.dependency_set.dependencies.keys().copied()
     }
 
-    fn super_head(&self, _chain: ChainId) -> Result<SuperHead, SupervisorError> {
-        todo!("implement call to ChainDbFactory")
+    fn super_head(&self, chain: ChainId) -> Result<SuperHead, SupervisorError> {
+        let db = self.database_factory.get_db(chain)?;
+
+        let l1_source = db.get_current_l1()?;
```

Comment:
why can't we just use `DerivationStorageReader::derived_to_source` passing the block id of the local unsafe block as param here @varun-doshi ?

Reply:
I think this comment is for me.

Reply:
I see, that is what I suspected, thanks for clarifying. will add this to docs for next person who wonders.

Reply:
https://github.com/op-rs/kona/pull/2127